RetailCoder.VBE/UI/Command/FindAllReferencesCommand.cs

```diff
@@ -163,6 +173,30 @@ private Declaration FindTarget(object parameter)
             return _state.FindSelectedDeclaration(_vbe.ActiveCodePane);
         }
 
+        private Declaration FindFormTarget()
+        {            
+            var project = _vbe.ActiveVBProject;
+            var component = _vbe.SelectedVBComponent;
+            
+            if (component != null && _vbe.SelectedVBComponent.HasDesigner)
+            {                
+                var designer = ((dynamic)component.Target).Designer; // COM Object, need to late bind
+                int selectedCount = designer.Selected.Count;
+
+                if (selectedCount > 1) { return null; }
+
+                var selectedType = selectedCount == 0 ? DeclarationType.ClassModule : DeclarationType.Control;
+                string selectedName = selectedCount == 0 ? component.Name : designer.Selected[0].Name;
+
+                return _state.DeclarationFinder.MatchName(selectedName)
+                                               .First(m => m.ProjectId == project.HelpFile
```

Comment:
`IVBProject.ProjectId` is an abstraction on top of `HelpFile` - let's avoid referring to `HelpFile` directly, so if/when we change the mechanics of `ProjectId`, we don't have a ton things falling apart because they were explicitly looking at `HelpFile`; please use `ProjectId` here.

Reply:
As an alternative, you could construct the `QualifiedModuleName` for the component and do an equality check on `m.QualifiedName.QualifiedModuleName` instead of checking the Helpfile and the name of the component. That would eliminate the need for the project entirely.